### DISCUSSION TITLE
If GNU components can be compiled by cmake

### DISCUSSION THREAD

**Discussion by RainbowTaro2022** [microsoft/vcpkg#34707] [Show and tell]
Originally, GNU components were difficult to compile into the Windows version, and sometimes they had to be compiled in other systems to do so

I have generated CMakeLists for some GNU components through Baidu's AI, and I'm not sure if you will adopt them

[Emacs Cmake.txt](https://github.com/microsoft/vcpkg/files/13115804/Emacs.Cmake.txt)
[Emms CMake.txt](https://github.com/microsoft/vcpkg/files/13115805/Emms.CMake.txt)
[Findutils CMake.txt](https://github.com/microsoft/vcpkg/files/13115807/Findutils.CMake.txt)
[GCC CMake.txt](https://github.com/microsoft/vcpkg/files/13115808/GCC.CMake.txt)
[Gcompris CMake（部分）.txt](https://github.com/microsoft/vcpkg/files/13115810/Gcompris.CMake.txt)
[Gettext CMake.txt](https://github.com/microsoft/vcpkg/files/13115811/Gettext.CMake.txt)
[gmp CMake.txt](https://github.com/microsoft/vcpkg/files/13115812/gmp.CMake.txt)
[Gnucash CMake.txt](https://github.com/microsoft/vcpkg/files/13115813/Gnucash.CMake.txt)
[gnupg CMake.txt](https://github.com/microsoft/vcpkg/files/13115814/gnupg.CMake.txt)
[GnuRadio CMake.txt](https://github.com/microsoft/vcpkg/files/13115815/GnuRadio.CMake.txt)
[grub CMake（部分）.txt](https://github.com/microsoft/vcpkg/files/13115816/grub.CMake.txt)
[Libiconv CMake.txt](https://github.com/microsoft/vcpkg/files/13115817/Libiconv.CMake.txt)
[Mpfr CMake.txt](https://github.com/microsoft/vcpkg/files/13115818/Mpfr.CMake.txt)
[Ncurses CMake.txt](https://github.com/microsoft/vcpkg/files/13115819/Ncurses.CMake.txt)

**Comment by Neumann-A** [comment]
vcpkg prefers upstreams buildsystem before vendoring a CMakeLists.txt. Additional half of the libraries are already in vcpkg?
Furthermore you are missing a lot of the logic of the upstream build.

**Reply by RainbowTaro2022** [reply]
What is the so-called upstream?
Is it the mirror source?

**Comment by dg0yt** [comment]
We should be glad that you chose some packages which already have ports in vcpkg. 
It helps to make clear that this uninformed AI output is useless.
